backward.hpp

```diff
@@ -746,7 +746,16 @@ class TraceResolverLinuxImpl<trace_resolver_tag::backtrace_symbol>:
 		while (*funcname && *funcname != '(') {
 			funcname += 1;
 		}
-		trace.object_filename.assign(filename, funcname++);
+		trace.object_filename.assign(filename, funcname); // it is ok if funcname is the ending \0, then we select string till end
+
+		if ( ! (*funcname) ) { // we already hit end of string. This happens for the last address 0xffff for ip==0
+			trace.object_function = "(none)";
+			trace.source.function = "(none)";
+			return trace;
+		}
+
+		// else normal string, we are at the opening '(' now
+		funcname++;
```

Comment:
Wrapping the `*funcname` should be enough. Hopefully more readable. This is the job of the stacktrace pretty printer to react on an empty `object_function`.

```cpp
if (*funcname) {
		char* funcname_end = funcname + 1;
		while (*funcname_end && *funcname_end != ')' && *funcname_end != '+') {
			funcname_end += 1;
		}
		*funcname_end = '\0';
		trace.object_function = this->demangle(funcname);
}
```

should be enough to prevent the off by one error.